genienlp/data_utils/database_utils.py

```diff
@@ -0,0 +1,106 @@
+import unicodedata
+import re
+
+import nltk
+
+nltk.download('stopwords', quiet=True)
+from nltk.corpus import stopwords
+
+
+DOMAIN_TYPE_MAPPING = dict()
```

Comment:
Why do you need this two-level mapping instead of mapping directly to ThingTalk types in the earlier function?
In any case, this cannot be hardcoded.

Reply:
Mainly for transparency. I wanted to keep everything in Q-space and only map to TT types in the final stage.
Also, in the previous embedding-based approach, bootleg_post_process_types is optional and we can just look up type Qids in type2id mapping and embed them.